pkg/disk/cloud.go

```diff
@@ -222,7 +222,7 @@ func attachDisk(ctx context.Context, tenantUserUID, diskID, nodeID string, isSha
 		devicePaths := calcNewDevices(before, after)
 
 		// BDF Disk Logical
-		if IsVFNode() && len(devicePaths) == 0 {
+		if !GlobalConfigVar.ControllerService && IsVFNode() && len(devicePaths) == 0 {
```

Comment:
After #993 , We can actually enable controller and node service in the same process. And using global variable is not good. Maybe we should introduce a new parameter to attachDisk(). We should replace all ADControllerEnable related branch with the new parameter.